query-engine/core/src/telemetry/capturing/capturer.rs

```diff
@@ -292,18 +266,16 @@ mod task {
         trace_id: TraceId,
     ) -> Result<Storage, tokio::sync::oneshot::error::RecvError> {
         let (sender, mut receiver) = oneshot::channel::<Storage>();
-        TASK.send(CaptureOp::FetchCaptures(trace_id, sender)).unwrap();
+        SENDER.send(CaptureOp::FetchCaptures(trace_id, sender)).unwrap();
         let mut interval = interval(Duration::from_millis(1));
-        let mut i = 0;
+        let start = std::time::Instant::now();
         loop {
             tokio::select! {
-                _ = interval.tick() => {
-                    i+=1;
-                },
                 msg = &mut receiver => {
-                    tracing::trace!("waited {}ms to receive signal from fetch captures", i);
+                    tracing::trace!("waited {}ms to receive signal from fetch captures", start.elapsed().as_millis());
                     return msg;
                 }
+                _ = interval.tick() => {},
             }
```

Comment:
Think we can even remove the loop and the `select!()`, it looks like the interval doesn't serve any purpose anymore.

Reply:
If I'm not wrong. If we don't wait it can be that the channel is not ready yet when we poll it which leads to an error, this assumes polling at most every ms. Which I think it's reasonable not to account for a higher latency per request.

Reply:
The `receiver.await` in my example in the previous thread takes care of that, the receiver itself is a future.

Reply:
Ok, will try and follow up. Thank you.

Reply:
Addressed in https://github.com/prisma/prisma-engines/pull/3664